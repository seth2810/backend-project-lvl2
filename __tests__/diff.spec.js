import genDiff from '../src/index.js';

describe.each(['json', 'yml', 'ini'])('%s', (extension) => {
  describe('flat', () => {
    it('should return correct diff', () => {
      const pathToFile1 = resolveFixturePath(`${extension}/flat/before.${extension}`);
      const pathToFile2 = resolveFixturePath(`${extension}/flat/after.${extension}`);
      const result = genDiff(pathToFile1, pathToFile2);

      expect(result).toMatchSnapshot();
    });
  });
});
