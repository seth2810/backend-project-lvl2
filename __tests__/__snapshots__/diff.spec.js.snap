// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`diff ini json should return correct diff 1`] = `"[{\\"children\\":[{\\"value\\":false,\\"type\\":\\"added\\",\\"key\\":\\"follow\\"},{\\"value\\":\\"Value 1\\",\\"type\\":\\"equals\\",\\"key\\":\\"setting1\\"},{\\"value\\":\\"200\\",\\"type\\":\\"removed\\",\\"key\\":\\"setting2\\"},{\\"left\\":true,\\"right\\":{\\"key\\":\\"value\\"},\\"type\\":\\"changed\\",\\"key\\":\\"setting3\\"},{\\"value\\":\\"blah blah\\",\\"type\\":\\"added\\",\\"key\\":\\"setting4\\"},{\\"value\\":{\\"key5\\":\\"value5\\"},\\"type\\":\\"added\\",\\"key\\":\\"setting5\\"},{\\"children\\":[{\\"value\\":\\"value\\",\\"type\\":\\"equals\\",\\"key\\":\\"key\\"},{\\"value\\":\\"vops\\",\\"type\\":\\"added\\",\\"key\\":\\"ops\\"}],\\"type\\":\\"nested\\",\\"key\\":\\"setting6\\"}],\\"type\\":\\"nested\\",\\"key\\":\\"common\\"},{\\"children\\":[{\\"left\\":\\"bas\\",\\"right\\":\\"bars\\",\\"type\\":\\"changed\\",\\"key\\":\\"baz\\"},{\\"value\\":\\"bar\\",\\"type\\":\\"equals\\",\\"key\\":\\"foo\\"},{\\"left\\":{\\"key\\":\\"value\\"},\\"right\\":\\"str\\",\\"type\\":\\"changed\\",\\"key\\":\\"nest\\"}],\\"type\\":\\"nested\\",\\"key\\":\\"group1\\"},{\\"value\\":{\\"abc\\":\\"12345\\"},\\"type\\":\\"removed\\",\\"key\\":\\"group2\\"},{\\"value\\":{\\"fee\\":\\"100500\\"},\\"type\\":\\"added\\",\\"key\\":\\"group3\\"}]"`;

exports[`diff ini plain should return correct diff 1`] = `
"Property 'common.follow' was added with value: false
Property 'common.setting2' was deleted
Property 'common.setting3' was changed from true to [complex value]
Property 'common.setting4' was added with value: 'blah blah'
Property 'common.setting5' was added with value: [complex value]
Property 'common.setting6.ops' was added with value: 'vops'
Property 'group1.baz' was changed from 'bas' to 'bars'
Property 'group1.nest' was changed from [complex value] to 'str'
Property 'group2' was deleted
Property 'group3' was added with value: [complex value]"
`;

exports[`diff ini stylish should return correct diff 1`] = `
"{
    common: {
      + follow: false
        setting1: Value 1
      - setting2: 200
      - setting3: true
      + setting3: {
            key: value
        }
      + setting4: blah blah
      + setting5: {
            key5: value5
        }
        setting6: {
            key: value
          + ops: vops
        }
    }
    group1: {
      - baz: bas
      + baz: bars
        foo: bar
      - nest: {
            key: value
        }
      + nest: str
    }
  - group2: {
        abc: 12345
    }
  + group3: {
        fee: 100500
    }
}"
`;

exports[`diff json json should return correct diff 1`] = `"[{\\"children\\":[{\\"value\\":false,\\"type\\":\\"added\\",\\"key\\":\\"follow\\"},{\\"value\\":\\"Value 1\\",\\"type\\":\\"equals\\",\\"key\\":\\"setting1\\"},{\\"value\\":200,\\"type\\":\\"removed\\",\\"key\\":\\"setting2\\"},{\\"left\\":true,\\"right\\":{\\"key\\":\\"value\\"},\\"type\\":\\"changed\\",\\"key\\":\\"setting3\\"},{\\"value\\":\\"blah blah\\",\\"type\\":\\"added\\",\\"key\\":\\"setting4\\"},{\\"value\\":{\\"key5\\":\\"value5\\"},\\"type\\":\\"added\\",\\"key\\":\\"setting5\\"},{\\"children\\":[{\\"value\\":\\"value\\",\\"type\\":\\"equals\\",\\"key\\":\\"key\\"},{\\"value\\":\\"vops\\",\\"type\\":\\"added\\",\\"key\\":\\"ops\\"}],\\"type\\":\\"nested\\",\\"key\\":\\"setting6\\"}],\\"type\\":\\"nested\\",\\"key\\":\\"common\\"},{\\"children\\":[{\\"left\\":\\"bas\\",\\"right\\":\\"bars\\",\\"type\\":\\"changed\\",\\"key\\":\\"baz\\"},{\\"value\\":\\"bar\\",\\"type\\":\\"equals\\",\\"key\\":\\"foo\\"},{\\"left\\":{\\"key\\":\\"value\\"},\\"right\\":\\"str\\",\\"type\\":\\"changed\\",\\"key\\":\\"nest\\"}],\\"type\\":\\"nested\\",\\"key\\":\\"group1\\"},{\\"value\\":{\\"abc\\":12345},\\"type\\":\\"removed\\",\\"key\\":\\"group2\\"},{\\"value\\":{\\"fee\\":100500},\\"type\\":\\"added\\",\\"key\\":\\"group3\\"}]"`;

exports[`diff json plain should return correct diff 1`] = `
"Property 'common.follow' was added with value: false
Property 'common.setting2' was deleted
Property 'common.setting3' was changed from true to [complex value]
Property 'common.setting4' was added with value: 'blah blah'
Property 'common.setting5' was added with value: [complex value]
Property 'common.setting6.ops' was added with value: 'vops'
Property 'group1.baz' was changed from 'bas' to 'bars'
Property 'group1.nest' was changed from [complex value] to 'str'
Property 'group2' was deleted
Property 'group3' was added with value: [complex value]"
`;

exports[`diff json stylish should return correct diff 1`] = `
"{
    common: {
      + follow: false
        setting1: Value 1
      - setting2: 200
      - setting3: true
      + setting3: {
            key: value
        }
      + setting4: blah blah
      + setting5: {
            key5: value5
        }
        setting6: {
            key: value
          + ops: vops
        }
    }
    group1: {
      - baz: bas
      + baz: bars
        foo: bar
      - nest: {
            key: value
        }
      + nest: str
    }
  - group2: {
        abc: 12345
    }
  + group3: {
        fee: 100500
    }
}"
`;

exports[`diff yml json should return correct diff 1`] = `"[{\\"children\\":[{\\"value\\":false,\\"type\\":\\"added\\",\\"key\\":\\"follow\\"},{\\"value\\":\\"Value 1\\",\\"type\\":\\"equals\\",\\"key\\":\\"setting1\\"},{\\"value\\":200,\\"type\\":\\"removed\\",\\"key\\":\\"setting2\\"},{\\"left\\":true,\\"right\\":{\\"key\\":\\"value\\"},\\"type\\":\\"changed\\",\\"key\\":\\"setting3\\"},{\\"value\\":\\"blah blah\\",\\"type\\":\\"added\\",\\"key\\":\\"setting4\\"},{\\"value\\":{\\"key5\\":\\"value5\\"},\\"type\\":\\"added\\",\\"key\\":\\"setting5\\"},{\\"children\\":[{\\"value\\":\\"value\\",\\"type\\":\\"equals\\",\\"key\\":\\"key\\"},{\\"value\\":\\"vops\\",\\"type\\":\\"added\\",\\"key\\":\\"ops\\"}],\\"type\\":\\"nested\\",\\"key\\":\\"setting6\\"}],\\"type\\":\\"nested\\",\\"key\\":\\"common\\"},{\\"children\\":[{\\"left\\":\\"bas\\",\\"right\\":\\"bars\\",\\"type\\":\\"changed\\",\\"key\\":\\"baz\\"},{\\"value\\":\\"bar\\",\\"type\\":\\"equals\\",\\"key\\":\\"foo\\"},{\\"left\\":{\\"key\\":\\"value\\"},\\"right\\":\\"str\\",\\"type\\":\\"changed\\",\\"key\\":\\"nest\\"}],\\"type\\":\\"nested\\",\\"key\\":\\"group1\\"},{\\"value\\":{\\"abc\\":12345},\\"type\\":\\"removed\\",\\"key\\":\\"group2\\"},{\\"value\\":{\\"fee\\":100500},\\"type\\":\\"added\\",\\"key\\":\\"group3\\"}]"`;

exports[`diff yml plain should return correct diff 1`] = `
"Property 'common.follow' was added with value: false
Property 'common.setting2' was deleted
Property 'common.setting3' was changed from true to [complex value]
Property 'common.setting4' was added with value: 'blah blah'
Property 'common.setting5' was added with value: [complex value]
Property 'common.setting6.ops' was added with value: 'vops'
Property 'group1.baz' was changed from 'bas' to 'bars'
Property 'group1.nest' was changed from [complex value] to 'str'
Property 'group2' was deleted
Property 'group3' was added with value: [complex value]"
`;

exports[`diff yml stylish should return correct diff 1`] = `
"{
    common: {
      + follow: false
        setting1: Value 1
      - setting2: 200
      - setting3: true
      + setting3: {
            key: value
        }
      + setting4: blah blah
      + setting5: {
            key5: value5
        }
        setting6: {
            key: value
          + ops: vops
        }
    }
    group1: {
      - baz: bas
      + baz: bars
        foo: bar
      - nest: {
            key: value
        }
      + nest: str
    }
  - group2: {
        abc: 12345
    }
  + group3: {
        fee: 100500
    }
}"
`;
